packages/dds/tree/src/test/simple-tree/objectNode.spec.ts

```diff
@@ -82,6 +85,71 @@ const schemaFactory = new SchemaFactory("Test");
 	}
 }
 
+// FieldHasDefault
+{
+	class Note extends schemaFactory.object("Note", {}) {}
+
+	// Node schema via ImplicitAllowedTypes
+	{
+		// Implicitly required field does not have a default value.
+		type _check = requireFalse<FieldHasDefault<typeof Note>>;
+	}
+
+	// Required field
+	{
+		type RequiredNoteField = FieldSchema<FieldKind.Required, typeof Note>;
+
+		// Required field does not have a default value.
+		type _check = requireFalse<FieldHasDefault<RequiredNoteField>>;
+	}
+
+	// Optional field
+	{
+		type OptionalNoteField = FieldSchema<FieldKind.Optional, typeof Note>;
+
+		// Optional field has default: undefined.
+		type _check = requireTrue<FieldHasDefault<OptionalNoteField>>;
+	}
+
+	// Identifier field
+	{
+		type IdentifierField = FieldSchema<FieldKind.Identifier, typeof SchemaFactory.string>;
+
+		// Identifier fields have default: undefined.
```

Comment:
not true. They generate a default identifier when hydrated.

Reply:
Ah, good point. I was thinking in terms of the inputs. Will update.